Minor improvements with test harness

mjp41 opened the pull request:
Output what is being done, so that it is easier to reproduce outside the test harness.

rengolin commented:
Doesn't ctest already have a verbose flag that shows the command lines? Or because some of those things are CMake related, they don't show?

I'd prefer if more verbose output could be hidden behind a verbose flag, for example `-DVERBOSE_TESTS=True` (possibly also setting ctest's verbose flag?).

mjp41 commented:
This will only print if you call ctest with  `-V` or `--output-on-failure` for failed tests.  Prior to this commit, it would dodo 
```
$ ctest -R hygienic-id -E txt -V
UpdateCTestConfiguration  from :/home/mjp41/src/verona/build-ninja/DartConfiguration.tcl
UpdateCTestConfiguration  from :/home/mjp41/src/verona/build-ninja/DartConfiguration.tcl
Test project /home/mjp41/src/verona/build-ninja
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 73
    Start 73: verona-parser/hygienic-id.out

73: Test command: /usr/bin/cmake "-DTESTFILE=hygienic-id.verona" "-DWORKING_DIR=/home/mjp41/src/verona/testsuite/verona-parser/" "-DTOOLCMD=/home/mjp41/src/verona/build-ninja/dist/verona-parser" "-DTOOLNAME=verona-parser" "-DOUTPUT_DIR=/home/mjp41/src/verona/build-ninja/testsuite/verona-parser/hygienic-id.out" "-P" "/home/mjp41/src/verona/testsuite/run_command.cmake"
73: Test timeout computed to be: 10000000
1/1 Test #73: verona-parser/hygienic-id.out ....   Passed    0.05 sec

The following tests passed:
        verona-parser/hygienic-id.out

100% tests passed, 0 tests failed out of 1

Total Test time (real) =   0.09 sec
```
But what the cmake actually does it hidden, so you can't directly run the command.  With this commit you get
```
UpdateCTestConfiguration  from :/home/mjp41/src/verona/build-ninja/DartConfiguration.tcl
UpdateCTestConfiguration  from :/home/mjp41/src/verona/build-ninja/DartConfiguration.tcl
Test project /home/mjp41/src/verona/build-ninja
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 73
    Start 73: verona-parser/hygienic-id.out

73: Test command: /usr/bin/cmake "-DTESTFILE=hygienic-id.verona" "-DWORKING_DIR=/home/mjp41/src/verona/testsuite/verona-parser/" "-DTOOLCMD=/home/mjp41/src/verona/build-ninja/dist/verona-parser" "-DTOOLNAME=verona-parser" "-DOUTPUT_DIR=/home/mjp41/src/verona/build-ninja/testsuite/verona-parser/hygienic-id.out" "-P" "/home/mjp41/src/verona/testsuite/run_command.cmake"
73: Test timeout computed to be: 10000000
73: Running
73:    /home/mjp41/src/verona/build-ninja/dist/verona-parser hygienic-id.verona --ast --validate
73: in working directory
73:    /home/mjp41/src/verona/testsuite/verona-parser/
73: output sent to
73:    /home/mjp41/src/verona/build-ninja/testsuite/verona-parser/hygienic-id.out
1/1 Test #73: verona-parser/hygienic-id.out ....   Passed    0.05 sec

The following tests passed:
        verona-parser/hygienic-id.out

100% tests passed, 0 tests failed out of 1

Total Test time (real) =   0.08 sec
```
This makes it easier to view the actual command being used.  Without `-V` you get
```
$ ctest -R hygienic-id -E txt
Test project /home/mjp41/src/verona/build-ninja
    Start 73: verona-parser/hygienic-id.out
1/1 Test #73: verona-parser/hygienic-id.out ....   Passed    0.05 sec

100% tests passed, 0 tests failed out of 1

Total Test time (real) =   0.08 sec
```
